packages/jest-circus/src/testCaseReportHandler.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import type {Circus} from '@jest/types';
+import {makeSingleTestResult, parseSingleTestResult} from './utils';
+
+const testCaseReportHandler = (
+  testPath: string,
+  sendMessageToJest: (eventName: string, args: Array<unknown>) => unknown,
+) => (event: Circus.Event) => {
+  switch (event.name) {
+    case 'test_done': {
+      const testResult = makeSingleTestResult(event.test);
+      const testCaseResult = parseSingleTestResult(testResult);
+      sendMessageToJest('test-case-result', [testPath, testCaseResult]);
```

Comment:
should we do the `deepCyclicCopy` here instead?

Reply:
Tried this, and both the snippets below show a leak

1.
```typescript
const testResult = makeSingleTestResult(event.test);
const testCaseResult = parseSingleTestResult(testResult);
sendMessageToJest(
   'test-case-result',
    deepCyclicCopy([testPath, testCaseResult], {keepPrototype: false}),
);
```

2.
```typescript
const testResult = makeSingleTestResult(event.test);
const testCaseResult = parseSingleTestResult(
  deepCyclicCopy(testResult, {keepPrototype: false}),
);
sendMessageToJest('test-case-result', [testPath, testCaseResult]);
break;
```

Reply:
aight, thanks for checking